modules/rtpengine/rtpengine.c

```diff
@@ -1957,12 +2027,23 @@ static bencode_item_t *rtpp_function_call(bencode_buffer_t *bencbuf, struct sip_
 			}
 			LM_ERR("proxy replied with error: %.*s\n", error.len, error.s);
 		}
-        goto error;
+		goto error;
 	}
 
 	if (body_out)
 		*body_out = body;
 
+	if (op == OP_DELETE) {
+		/* Delete the key<->value from the hashtable */
+		if (!rtpengine_hash_table_remove(&callid)) {
```

Comment:
I don't think immediately deleting the hash table entry on "delete" is a good idea as it may only be a single branch that has been deleted and a "delete" does not preclude the possibility of more commands (offer, answer, or something else) coming for the same call-ID. Perhaps resetting the expiry value for the entry to something small might be a better idea.